### ISSUE TITLE
[Bug?]: url search parameter is erased if originated from outside the app (via redirection)

### ISSUE THREAD

**Issue by taivo** [redwoodjs/redwood#9816]
### What's not working?

My specific usecase is DiscourseConnect (have the redwood app handle authentication for the discourse forum). Discourse would redirect to a predefined route in the redwood app, with the URL shaped like this `https://my-redwood-site.com/sso?sso=PAYLOAD&sig=SIG`.

In the Redwoodjs page that serves as the SSO provider, both the `sso` and `sig` params are accessible in the page's code. However, the url displayed in the browser is stripped of the entire search query.

The issue is not limited to the DiscourseConnect usecase. Just paste a valid Redwood app url with a search query to the browser. Redwood will load that page and erase the search query from the browser location bar.

### How do we reproduce the bug?

Route.ts
```
<Route path="/test" page={TestPage} name="test" />
```

TestPage.tsx
```
const TestPage = ({ p1, p2 }) => {
  return (
    <>
      <div>p1: {p1}</div>
      <div>p2: {p2}</div>
    </>
  )
}
```

Start your dev server, and paste `http://localhost:8910/test?p1=hello&p2=world` into the browser. The TestPage component will render as expected, but the browser's location bar will only show `http://localhost:8910/test`.

Now from another page, redirect to the TestPage
```
<Redirect to={routes.test({p1: "Sup", p2: "friend"})} />
```

For this second case, the location bar retains the search string. The pasted URL usecase should also behave like this.

### What's your environment? (If it applies)

```shell
Chrome browser. 
RedwoodJS 6.6.0
```

### Are you interested in working on this?

- [ ] I'm interested in working on this

**Comment by ahaywood** [comment]
Hey @taivo 👋 I created a simple Redwood app, trying to replicate the problem that you're experiencing and I'm having some trouble.

Here's a Loom video that I recorded walking through my repo (sorry for the snoring dog in the background 🙉 ) : https://www.loom.com/share/0bac4127c13a4c309c94707749045443

Here's my sample repo: https://github.com/ahaywood/redwood-test-url-params

Would you mind to take a look? Maybe I'm missing something? Or maybe you could try sharing your repo?

**Comment by taivo** [comment]
Hi @ahaywood , thank you for looking into this so quickly.

TLDR: it turns out this is specific to the supabase auth adapter. The culprit is https://github.com/redwoodjs/redwood/blob/main/packages/auth-providers/supabase/web/src/supabase.ts#L208-L211

That seems to come out of nowhere, right? Here's how I arrived at the conclusion above:

- First, I cloned the test repo you created and confirmed that it does not have the problem I described. 
- 2nd, I started removing stuff from my project to move it toward something equivalent to yours, hoping the problem goes away somewhere along the way. It did when I removed `<AuthProvider>` from `App.tsx`
- 3rd, I added the following hack at the top of `App.tsx` to see what is triggering the url change
```
;(function (history) {
  console.log('WE ARE OVERRIDING REPLACE_STATE')
  const replaceState = history.replaceState
  history.replaceState = function (state) {
    console.log('REPLACE_STATE CALLED')

    debugger
    return replaceState.apply(history, arguments)
  }
})(window.history)
```
- Finally, I pasted a URL with search parameters. When the chrome debugger kicks in, I see this in the call stack:
![image](https://github.com/redwoodjs/redwood/assets/1006076/701a199e-2bfc-4182-b61a-f0db3e344452)


I'm guessing that what's happening here is the entire redwood app is being initialized when we paste an arbitrary app URL, and the supabase adapter gets initialized during that process. There may be cases where the current behavior is actually the correct behavior.

Also, it's worth noting that the problem I've described in this issue occurs both when a user is already logged in and not.

**Comment by ahaywood** [comment]
@taivo Thanks for digging a little deeper. I'm going to follow up with some of the people on the team that have specialized in auth and see if they've run into something similar. ... Stay tuned!
